src/thunderbots/software/ai/world/team.h

```diff
@@ -20,22 +21,62 @@ class Team
    public:
     /**
      * Create a new team
+     *
+     * @param robot_expiry_buffer_milliseconds The number of milliseconds a robot must not
+     * have been updated for before it is removed from the team
      */
-    explicit Team();
+    explicit Team(const std::chrono::milliseconds robot_expiry_buffer_milliseconds);
 
     /**
      * Updates this team with new robots.
      *
      * @param team_robots the new robots for this team
+     * @param timestamp The timestamp at which this update is taking place. The timestamp
+     * must be >= the last update timestamp of the robots on the team
+     */
+    void updateRobots(const std::vector<Robot>& team_robots,
+                      const std::chrono::steady_clock::time_point timestamp);
+
+    /**
+     * Updates this team with new data from the given team object
+     *
+     * @param new_team_data A team with the new team data
+     * @param timestamp The timestamp at which this update is taking place. The timestamp
+     * must be >= the last update timestamp of the robots on the team
+     */
+    void updateState(const Team& new_team_data,
+                     const std::chrono::steady_clock::time_point timestamp);
+
+    /**
+     * Updates the Team's state to be its predicted state at the given timestamp.
+     * The timestamp must be >= the last update timestamp of the robots on the team
+     *
+     * @param timestamp The timestamp at which to update the team's state to. Must
+     * be >= the last update timestamp of the robots on the team
+     */
+    void updateStateToPredictedState(
+        const std::chrono::steady_clock::time_point timestamp);
+
+    /**
+     * Removes expired robots from the team. Robots are expired if it has been more than
+     * the expiry_buffer time has passed since they were last updated.
+     *
+     * @param timestamp The timestamp for when this removal is taking place
      */
-    void update(std::vector<Robot> &team_robots);
+    void removeExpiredRobots(const std::chrono::steady_clock::time_point timestamp);
```

Comment:
Might be useful to mention in the docstring under what real-world conditions a robot would be considered "expired"

Reply:
done